e2e/internal/e2e/keyexec.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package e2e
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+var importPrivKeyScript string
+
+func getS(kpath string) string {
+	// Yes, this uses /usr/bin/expect
+	return fmt.Sprintf(`
+set timeout -1
+
+set psk "e2etests"
+
+spawn singularity key import %s
+
+expect "Enter your old password : "
+send "${psk}\r"
+
+expect "Enter a new password for this key : "
+send "${psk}\r"
+
+expect "Retype your passphrase : "
+send "${psk}\r"
+
+expect eof
+`, kpath)
+}
+
+func getE(num int, kpath, armor, psk string) string {
+	// Yes, this uses /usr/bin/expect
+	return fmt.Sprintf(`
+set timeout -1
+
+set psk "e2etests"
+
+spawn singularity key export --secret %s %s
+
+expect "Enter # of signing key to use : "
+send "%v\r"
+
+expect "Enter key passphrase : "
+send "%s\r"
+
+expect eof
+`, armor, kpath, num, psk)
+}
+
+var backupSypgp = filepath.Join(HomeDir(), ".singularity/sypgp/secret-keyring-backup")
+
+// PullDefaultPublicKey will pull the public Sylabs Admin key
+func PullDefaultPublicKey(t *testing.T) {
+	LoadEnv(t, &testenv)
+
+	argv := []string{"key", "pull", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
+
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	out, err := execKey.CombinedOutput()
+	if err != nil {
+		t.Log(string(out))
+		t.Fatalf("Unable to pull key: %v", err)
+	}
+}
+
+// RemoveDefaultPublicKey will pull the public Sylabs Admin key
+func RemoveDefaultPublicKey(t *testing.T) {
+	LoadEnv(t, &testenv)
+
+	argv := []string{"key", "remove", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	out, err := execKey.CombinedOutput()
+	if err != nil {
+		t.Log(string(out))
+		t.Fatalf("Unable to pull key: %v", err)
+	}
+}
+
+// BackupSecretKeyring ...
+func BackupSecretKeyring(t *testing.T) {
+	err := os.Rename(filepath.Join(HomeDir(), ".singularity/sypgp/pgp-secret"), backupSypgp)
+	if err != nil {
+		t.Fatalf("Unable to rename secret keyring: %v", err)
+	}
+}
+
+// RecuverSecretKeyring ...
+func RecuverSecretKeyring(t *testing.T) {
```

Comment:
What does this function do? @WestleyK

Reply:
This function `RecoverSecretKeyring` will recover your old keyring. When the tests run you keyring is backup to a tmp file, after the tests run your keyring is recovered.

Reply:
okay good, there's a typo in this function name